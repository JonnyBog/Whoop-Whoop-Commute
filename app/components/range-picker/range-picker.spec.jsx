import React from 'react';
import renderer from 'react-test-renderer';

import RangePicker from 'components/range-picker/range-picker';

describe('Components', () => {
  describe('RangePicker', () => {
    it('should export', () => {
      expect(RangePicker).toBeDefined();
    });

    let props;

    beforeEach(() => {
      props = {
        max: 5,
        label: 'test',
        id: 'test',
        value: 1,
        onChange: jest.fn()
      };
    });

    afterEach(() => {
      props = null;
    });

    /**
     * createRenderTree
     * @param   {Object} updatedProps - defaults to props
     * @returns {JSX} component tree
     */
    function createRenderTree (updatedProps = props) {
      return renderer.create(
        <RangePicker {...updatedProps} />
      ).toJSON();
    }

    describe('Appearance', () => {
      it('should render', () => {
        const tree = createRenderTree();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
