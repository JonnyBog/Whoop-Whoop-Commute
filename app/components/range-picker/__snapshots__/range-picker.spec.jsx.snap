// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components RangePicker Appearance should render 1`] = `
Array [
  <p
    className="range-picker__RangePickerLabel-dqh1ap-0 hJQUTf"
  >
    test
  </p>,
  <input
    id="test"
    max={5}
    min="1"
    onChange={[MockFunction]}
    type="range"
    value="test"
  />,
]
`;
